code/game/objects/items/devices/pipe_painter.dm

```diff
@@ -19,10 +20,10 @@
 
 /obj/item/device/pipe_painter/afterattack(atom/A, mob/user, proximity_flag)
 	//Make sure we only paint adjacent items
-	if(proximity_flag!= 1)
+	if(!proximity_flag)
 		return
 
-	if(!istype(A,/obj/machinery/atmospherics/pipe/simple) && !istype(A,/obj/machinery/atmospherics/pipe/manifold) && !istype(A,/obj/machinery/atmospherics/pipe/manifold4w))
+	if(!istype(A,/obj/machinery/atmospherics/pipe))
```

Comment:
you can now paint heat exchange pipes! (except you could already do it with the pipe_dispenser's paint mode...)